### ISSUE TITLE
Request all elements to have appropriate accessible names and roles assigned (accessibility)

### ISSUE THREAD

**Issue by robeastwood-agency** [aehrc/smart-forms#1735]
**Is your feature request related to a problem? Please describe.**
Raised by accessibility experts
> All elements have appropriate accessible names and roles assigned.
> Some interactive elements do not consistently expose their name, role, or value, making them difficult to identify and operate for users relying on assistive technologies.
> For example:
> * The menu of assessment forms visually resembles clickable cards but is not announced as links or buttons by screen readers.
> * The “Clear” button (an “X” icon) in text input fields is inconsistently operable and announced. In the “Height” field, it is announced only as “button,” without a descriptive name.

**Describe the solution you'd like**
Recommendation by accessibility experts:
> * Ensure use of semantic HTLM and expose roles programmatically. Be sure to provide descriptive accessible names. 
> * Ensure consistency across elements ensuring that text input fields have a "clear" button where appropriate and is keyboard accessible. Be sure to use aria-label to ensure that elements are announced correctly. 

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by fongsean** [comment]
Hi @robeastwood-agency, I have pushed a potential fix on v1.2.10.

I actually wasn't able to tab to the (x) button no matter what I do, with or without VoiceOver, and tried it on different versions of the renderer. I wasn't able to tab to the item headings as well (might be I don't have the most updated CHAP form).

Hence, I made the change blindly by removing a wrapper <span> and adding aria-hidden=true to the button. In theory that should prevent the button from being "tabbed" to. Can you try it out on your end?

**Comment by MaryamMehd** [comment]
Hi,

Also checking any updates on this issue.

Thanks!
